sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala

```diff
@@ -74,11 +76,13 @@ object ResolveHints {
           case ResolvedHint(u: UnresolvedRelation, hint)
               if relations.exists(resolver(_, u.tableIdentifier.table)) =>
             relations.remove(u.tableIdentifier.table)
-            ResolvedHint(u, createHintInfo(hintName).merge(hint, handleOverriddenHintInfo))
+            ResolvedHint(u,
+              createHintInfo(hintName).merge(hint, hintErrorHandler.handleHintOverridden))
```

Comment:
nit: can we just pass the `hintErrorHandler` to `merge`? That seems more flexible.

Reply:
Yeah, that makes sense.